test/oxo.js

```diff
@@ -0,0 +1,345 @@
+    const OWNER = accounts[0];
+    let oxo;
+    let defaultBet = 1000;
+    let defaultGameId = 0;
+
+    before('setup', () => {
+        return OXO.deployed()
+            .then(instance => oxo = instance)
+            .then(reverter.snapshot);
+    });
+
+
+    it('should allow to create the game ', async () => {
+
+        const player = accounts[1];
+        let gameId = await oxo.createGame.call({from: player, value: defaultBet}).valueOf();
+        assert.equal(gameId, 0);
+    });
+
+    it('should deny to create the game with incorrect bet', async () => {
+
+        const player = accounts[1];
+        var res = await asserts.throws(oxo.createGame({from: player, value: 0}));
+
+        const value = '0xffffffsffffffffffffffffffffffffffffffffffffffffffffffffffffffffff';
+        var res = await asserts.throws(oxo.createGame({from: player, value: value}));
+
+
+        await asserts.throws(oxo.createGame({from: player, value: value / 2}));
+    });
+
+    it('should emit event about game creation ', async () => {
+
+        const player = accounts[1];
+        var result = await oxo.createGame({from: player, value: defaultBet});
+        assert.equal(result.logs[0].event, 'GameCreated');
+        assert.equal(result.logs[0].args.gameId.valueOf(), defaultGameId);
+        assert.equal(result.logs[0].args.bet.valueOf(), defaultBet);
+    });
+
+    it('should allow user to join the game', async () => {
+
+        const player1 = accounts[1];
+        const player2 = accounts[2];
+        await oxo.createGame({from: player1, value: defaultBet});
+        let result = await oxo.joinGame(defaultGameId, {from: player2, value: defaultBet});
+        assert.equal(result.logs[0].event, 'GameStarted');
+        assert.equal(result.logs[0].args.gameId.valueOf(), defaultGameId);
+        assert.equal(result.logs[0].args.player.valueOf(), player2);
+    });
+
+    it('should deny user to join the game if there are 2 players already', async () => {
+
+        const player1 = accounts[1];
+        const player2 = accounts[2];
+        const player3 = accounts[3];
+        await oxo.createGame({from: player1, value: defaultBet});
+        await oxo.joinGame(defaultGameId, {from: player2, value: defaultBet});
+        await asserts.throws(oxo.joinGame(defaultGameId, {from: player3, value: defaultBet}));
+    });
+
+
+    it('should deny user to join the game if he bet different value', async () => {
+
+        const player1 = accounts[1];
+        const player2 = accounts[2];
+        await oxo.createGame({from: player1, value: defaultBet});
+        await asserts.throws(oxo.joinGame(defaultGameId, {from: player2, value: defaultBet + 1}));
+    });
+
+
+    it('should deny user to join the game if its not yet created', async () => {
+
+        const player2 = accounts[2];
+        await asserts.throws(oxo.joinGame(defaultGameId, {from: player2, value: defaultBet}));
+    });
+
+
+    it('should allow user to make move', async () => {
```

Comment:
This test also checks event emitting. How else can you check that a move was made? 
I mean, this test also should be but with different name. And you could add another one test which checks something else that make you sure that move has been made.

Reply:
i got ur point :)